astroquery/ipac/ned/tests/test_ned.py

```diff
@@ -47,12 +48,14 @@ def patch_get(request):
 
 @pytest.fixture
 def patch_get_readable_fileobj(request):
+    @contextmanager
     def get_readable_fileobj_mockreturn(filename, cache=True, encoding=None,
                                         show_progress=True):
         # Need to read FITS files with binary encoding: should raise error
         # otherwise
         assert encoding == 'binary'
-        return open(data_path(DATA_FILES['image']), 'rb')
+        with open(data_path(DATA_FILES['image']), 'rb') as infile:
+            yield infile
```

Comment:
I guess it is okay to have a context manager inside a context manager, though such is not used in the official example usage at https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager . 🤷 

```suggestion
        infile = open(data_path(DATA_FILES['image']), 'rb')
        try:
            yield infile
        finally:
            infile.close()
```

Reply:
I recall I've seen it somewhere, maybe in the pytest or mock docs or a very similar SO problem. I would rather keep it than switch to try/finally and manual close.